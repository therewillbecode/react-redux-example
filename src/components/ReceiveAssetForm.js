import React from "react";
import { Form, Label, Segment } from "semantic-ui-react";

const ReceiveAssetForm = ({
  handleSubmit,
  handleNameChange,
  handleCommentChange,
  name,
  comment,
  nameErr
}) => (
  <div
    style={{
      width: "30%",
      left: "35%",
      position: "relative",
      paddingTop: "2em"
    }}
  >
    <Segment inverted color="black">
      <Form onSubmit={handleSubmit} inverted>
        <h2 style={{ textAlign: "center" }}>Add Received Asset To Inventory</h2>
        <Form.Group inline>
          <Form.Field required error={nameErr !== null}>
            {nameErr ? <Label pointing="below">{nameErr}</Label> : null}
            <Form.Input
              inverted
              name="name"
              label="Name"
              placeholder="Name"
              value={name}
              onChange={handleNameChange}
            />
          </Form.Field>
          <Form.Field>
            <Form.TextArea
              label="Comments"
              placeholder="Add Comments..."
              name="Comment"
              value={comment}
              onChange={handleCommentChange}
            />
          </Form.Field>
          <Form.Button primary content="Submit" />
        </Form.Group>
      </Form>
    </Segment>
  </div>
);

export default ReceiveAssetForm;
