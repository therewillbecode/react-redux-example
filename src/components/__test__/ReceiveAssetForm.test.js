/* eslint-disable */
import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";
import sinon from "sinon";
import { shallow } from "enzyme";
import { Form } from "semantic-ui-react";

import ReceiveAssetForm from "../ReceiveAssetForm";

describe("ReceiveAssetForm", () => {
  it("renders correctly", () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<ReceiveAssetForm />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});

describe("Calls handler props", () => {
  let spy = null;

  beforeEach(() => {
    spy = new sinon.spy();
  });

  it("Input should call onChange prop with correct args", () => {
    const expectedArg = { name: "name", value: "Foo" };
    const wrapper = shallow(<ReceiveAssetForm onChange={spy} />);
    wrapper.find(Form.Input).simulate("change", { target: expectedArg });
    const { target } = spy.firstCall.args[0];

    expect(target).toEqual(expectedArg);
  });

  it("TextArea should call onChange prop with correct args", () => {
    const expectedArg = { name: "comment", value: "Lorem Ipsum" };
    const wrapper = shallow(<ReceiveAssetForm onChange={spy} />);
    wrapper.find(Form.TextArea).simulate("change", { target: expectedArg });
    const { target } = spy.firstCall.args[0];

    expect(target).toEqual(expectedArg);
  });

  it("should call handleSubmit prop when form is submitted", () => {
    const wrapper = shallow(<ReceiveAssetForm handleSubmit={spy} />);
    wrapper.find(Form).simulate("submit");
    expect(spy.calledOnce).toBe(true);
  });
});
