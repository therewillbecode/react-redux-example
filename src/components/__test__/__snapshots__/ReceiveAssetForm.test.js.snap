// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReceiveAssetForm renders correctly 1`] = `
<div
  style={
    Object {
      "left": "35%",
      "paddingTop": "2em",
      "position": "relative",
      "width": "30%",
    }
  }
>
  <Segment
    color="grey"
    inverted={true}
  >
    <Form
      as="form"
      inverted={true}
      onSubmit={undefined}
    >
      <h2
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Add Received Asset To Inventory
      </h2>
      <FormGroup
        inline={true}
      >
        <FormField
          error={true}
          required={true}
        >
          <FormInput
            as={[Function]}
            control={[Function]}
            inverted={true}
            label="Name"
            name="name"
            onChange={undefined}
            placeholder="Name"
            value={undefined}
          />
        </FormField>
        <FormField>
          <FormTextArea
            as={[Function]}
            control={[Function]}
            label="Comments"
            name="Comment"
            onChange={undefined}
            placeholder="Add Comments..."
            value={undefined}
          />
        </FormField>
        <FormButton
          as={[Function]}
          content="Submit"
          control={[Function]}
          primary={true}
        />
      </FormGroup>
    </Form>
  </Segment>
</div>
`;
